### PR TITLE
add user profile command to context menu

### PR DESCRIPTION
**Pull request by FoxLisk**

### CODE REVIEW

File: src/discord/application_command_definitions.rs [hunk 1]
```diff
@@ -60,13 +61,15 @@ pub fn application_command_definitions() -> Vec<Command> {
...
-        })
+    .option(CommandOption {
+        description:
+            "A message (such as a rando permalink) to show the racers when they click Start"
+                .to_string(),
+        kind: CommandOptionType::String,
+        name: "on_start_message".to_string(),
+        required: Some(false),
+        ..command_option_default()
+    })
```
Comment (FoxLisk) [inline src/discord/application_command_definitions.rs:72]: this is just automatic reformatting

File: src/discord/mod.rs [hunk 2]
```diff
@@ -221,13 +223,24 @@ pub struct ErrorResponse {
...
             internal_error: internal_error.to_string(),
         }
     }
+
+    fn new_internal(internal_error: impl Display) -> Self {
+        Self {
+            user_facing_error: "Internal error, sorry.".to_string(),
+            internal_error: internal_error.to_string(),
+        }
+    }
```
Comment (FoxLisk) [inline src/discord/mod.rs:243]: it would be good to do a sweep of replacing some annoying code with this

but actually i think the `ErrorResponse` thing has basically been a failure overall and it would be nice to come up with a better pattern entirely :\

File: src/discord/application_command_definitions.rs [hunk 3]
```diff
@@ -642,7 +645,11 @@ pub fn application_command_definitions() -> Vec<Command> {
     })
     .build();
 
+    // N.B. description _must_ be empty
+    let user_profile = CommandBuilder::new(USER_PROFILE_CMD, "", CommandType::User).build();
```
Comment (FoxLisk) [inline src/discord/application_command_definitions.rs:649]: this adds the new command to the context menu

File: src/discord/interaction_handlers/application_commands.rs [hunk 4]
```diff
@@ -563,6 +568,36 @@ async fn handle_update_user_info(
     }
 }
 
+async fn handle_user_profile(
```
Comment (FoxLisk) [inline src/discord/interaction_handlers/application_commands.rs:571]: and this function is the meat of the MR, it's where the actual responding happens

File: src/discord/interaction_handlers/application_commands.rs [hunk 5]
```diff
@@ -616,7 +657,7 @@ fn format_player(content: Option<String>, player: &Player) -> InteractionRespons
         fields,
         footer: None,
         image: None,
-        kind: "".to_string(),
+        kind: "rich".to_string(),
```
Comment (FoxLisk) [inline src/discord/interaction_handlers/application_commands.rs:660]: the value here doesn't appear to be respected at all, but it seems better to pass the blessed value

File: src/discord/interaction_handlers/application_commands.rs [hunk 6]
```diff
@@ -630,6 +671,7 @@ fn format_player(content: Option<String>, player: &Player) -> InteractionRespons
         data: Some(InteractionResponseData {
             content,
             embeds: Some(embeds),
+            flags: Some(MessageFlags::EPHEMERAL),
```
Comment (FoxLisk) [inline src/discord/interaction_handlers/application_commands.rs:674]: kind of a driveby that i learned about while figuring out how to do this - this kind of read-only status info is nicer if it only shows for the requester and doesn't clutter things up

File: src/web/auth.rs [hunk 7]
```diff
@@ -176,7 +176,7 @@ impl<'r> FromRequest<'r> for Admin {
                 let mut sm = sm_lock.lock().await;
                 let un = state
                     .get_user(uid)
-                    .map(|u| u.name)
+                    .map(|u| u.global_name.unwrap_or(u.name))
```
Comment (FoxLisk) [inline src/web/auth.rs:179]: this and the next change are drive bys to do with the discord username change (`user.name` is no longer the best option to choose for a default display name)

**Comment by FoxLisk** [comment]
fixes #115  btw

**Review by RequiemOfSpirit — APPROVED** [review]
